homeassistant/components/alarm_control_panel/device_trigger.py

```diff
@@ -0,0 +1,151 @@
+        if entry.domain != DOMAIN:
+            continue
+
+        entity_state = hass.states.get(entry.entity_id)
+
+        # We need a state or else we can't populate the HVAC and preset modes.
+        if entity_state is None:
+            continue
+
+        supported_features = entity_state.attributes["supported_features"]
+
+        # Add triggers for each entity that belongs to this integration
+        triggers += [
+            {
+                CONF_PLATFORM: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_ENTITY_ID: entry.entity_id,
+                CONF_TYPE: "disarmed",
+            },
+            {
+                CONF_PLATFORM: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_ENTITY_ID: entry.entity_id,
+                CONF_TYPE: "triggered",
+            },
+        ]
+        if supported_features & SUPPORT_ALARM_ARM_HOME:
+            triggers.append(
+                {
+                    CONF_PLATFORM: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "armed_home",
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_AWAY:
+            triggers.append(
+                {
+                    CONF_PLATFORM: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "armed_away",
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_NIGHT:
+            triggers.append(
+                {
+                    CONF_PLATFORM: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "armed_night",
+                }
+            )
+
+    return triggers
+
+
+async def async_attach_trigger(
+    hass: HomeAssistant,
+    config: ConfigType,
+    action: AutomationActionType,
+    automation_info: dict,
+) -> CALLBACK_TYPE:
+    """Attach a trigger."""
+    config = TRIGGER_SCHEMA(config)
+
+    if config[CONF_TYPE] == "triggered":
+        from_state = STATE_ALARM_PENDING
+        to_state = STATE_ALARM_TRIGGERED
+    elif config[CONF_TYPE] == "disarmed":
+        from_state = STATE_ALARM_TRIGGERED
+        to_state = STATE_ALARM_DISARMED
+    elif config[CONF_TYPE] == "armed_home":
+        from_state = STATE_ALARM_PENDING
+        to_state = STATE_ALARM_ARMED_HOME
+    elif config[CONF_TYPE] == "armed_away":
+        from_state = STATE_ALARM_PENDING
+        to_state = STATE_ALARM_ARMED_AWAY
+    elif config[CONF_TYPE] == "armed_night":
+        from_state = STATE_ALARM_PENDING
+        to_state = STATE_ALARM_ARMED_NIGHT
```

Comment:
Is the `from_state` required in this example? :)
I was not quite sure but I attempted to implemented everything from the [docs](https://www.home-assistant.io/integrations/manual/#state-machine).

Reply:
from state is important because else it will trigger if it goes from armed_night -> armed_night when an attribute changes.

Reply:
That would be really bad 😁 thanks for reviewing and merging my PR!